tinygrad/runtime/ops_cpu.py

```diff
@@ -12,7 +12,10 @@ def einsum_mulacc(einsum, get_strides, expand):
   def einscripts(x): return ''.join(["abcdefghijklmnopqrstuvwxyz"[i] for i in x])
   def axes_slice(strides): return tuple(i for i,s in enumerate(strides) if s != 0), tuple(slice(None) if s != 0 else 0 for s in strides)
   def mulacc(a, b, new_shape):
-    (a_axes, a_slices), (b_axes, b_slices) = axes_slice(get_strides(a)), axes_slice(get_strides(b))
+    (a_axes, a_slices), (b_axes, b_slices) = axes_slice(a_strides := get_strides(a)), axes_slice(get_strides(b))
```

Comment:
i'd remove the `axes_slice` helper since it's incomplete, and refactor the whole logic. also it's not obvious why it only updates `a` and not `b`

Reply:
- You are right for pointing out the need to refactor, though it should be within the `axes_slice` helper. The current code handles concerns for this helper outside of it, so I refactored it to take `sum_axes` into account, which is cleaner and less confusing.
- As for why only one operand is needed, this is because `np.einsum` handles broadcasting the value for the operand with missing axes. However, this isn't part of the new code, which is also cleaner and less confusing.